File: Network_topology/pythoncode/ring.py
import numpy as np
import matplotlib.pyplot as plt
import init

## 读取数据
r=init.Parameter()
x = np.load("x.npy")
y = np.load("y.npy")

## 绘制环型拓扑结构
fig = plt.figure(dpi=80)
plt.xlim(0,r.xlim)
plt.ylim(0,r.ylim)
# 绘制网络节点
plt.scatter(x,y)
# 成环
Point = np.stack([x,y])
d = np.array([x-np.mean(x),y-np.mean(y)])   # 各点到中点的连线
angle = np.arctan2(d[1,:],d[0,:])           # 各点中点连线与x轴角度
# 按角度排序
i = np.argsort(angle)
angle= np.sort(angle)
x = Point[0,i].tolist()
y = Point[1,i].tolist()
x.append(x[0]) #加入起点到最末，形成闭合图形
y.append(y[0])
plt.plot(x,y)

plt.show()

File: Network_topology/matlabcode/ring.m
%% 清空环境变量
clc;
clear;
close all;

%% 读取数据
r=load('data.mat');
x = r.x;
y = r.y;

%% 绘制环型拓扑结构
figure
axis([0 r.xlim 0 r.ylim]);
% 绘制网络节点
scatter(x,y);
hold on
% 成环
Point = [x;y];
d = [x-mean(x);y-mean(y)]; %各点到中点的连线
angle = atan2(d(2,:),d(1,:)); % 各点中点连线与x轴角度
% 按角度排序
[angle, i] = sort(angle);   
Point = Point(:,i); 
Point = [Point Point(:,1)]; %加入起点到最末，形成闭合图形
plot( Point(1,:), Point(2,:), '*-b');
